nni/algorithms/compression/v2/pytorch/utils/pruning.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from copy import deepcopy
+from typing import Dict, List
+
+from torch.nn import Module
+
+
+def config_list_canonical(model: Module, config_list: List[Dict]) -> List[Dict]:
+    '''
+    Split the config by op_names if 'sparsity' or 'sparsity_per_layer' in config,
+    and set the sub_config['total_sparsity'] = config['sparsity_per_layer'].
+    '''
+    for config in config_list:
+        if 'sparsity' in config:
+            if 'sparsity_per_layer' in config:
+                raise ValueError("'sparsity' and 'sparsity_per_layer' have the same semantics, can not set both in one config.")
+            else:
+                config['sparsity_per_layer'] = config.pop('sparsity')
+
+    config_list = dedupe_config_list(unfold_config_list(model, config_list))
+    new_config_list = []
+
+    for config in config_list:
+        if 'sparsity_per_layer' in config:
+            sparsity_per_layer = config.pop('sparsity_per_layer')
+            op_names = config.pop('op_names')
+            for op_name in op_names:
+                sub_config = deepcopy(config)
+                sub_config['op_names'] = [op_name]
+                sub_config['total_sparsity'] = sparsity_per_layer
+                new_config_list.append(sub_config)
+        elif 'max_sparsity_per_layer' in config and isinstance(config['max_sparsity_per_layer'], float):
```

Comment:
Why do we need isinstance(config['max_sparsity_per_layer'], float) here?

Reply:
If it is float, convert it to dict, skip otherwise.

Reply:
got it